File: src/main/java/de/saschahlusiak/freebloks/database/FreebloksDB.java
package de.saschahlusiak.freebloks.database;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteException;

public abstract class FreebloksDB {
	Context context;
	SQLiteDatabase db;
	FreebloksDBOpenHandler dbHelper;

	public FreebloksDB(Context context) {
		this.context = context;
	}

	public void open() throws SQLiteException {
		db = null;
		dbHelper = new FreebloksDBOpenHandler(context);
		db = dbHelper.getWritableDatabase();
	}

	public void close() {
		if (dbHelper != null)
			dbHelper.close();
	}

	public void beginTransaction() {
		db.beginTransaction();
	}

	public void endTransaction(boolean success) {
		if (success)
			db.setTransactionSuccessful();

		db.endTransaction();
	}
}

File: app/src/main/kotlin/de/saschahlusiak/freebloks/database/FreebloksDB.kt
package de.saschahlusiak.freebloks.database

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteException

abstract class FreebloksDB(private val context: Context) {
	protected var db: SQLiteDatabase? = null
    private var dbHelper: FreebloksDBOpenHandler? = null

    @Throws(SQLiteException::class)
    fun open() {
        dbHelper = FreebloksDBOpenHandler(context)
        db = dbHelper?.writableDatabase
    }

    fun close() {
        dbHelper?.close()
        dbHelper = null
    }

    fun beginTransaction() {
        db?.beginTransaction()
    }

    fun endTransaction(success: Boolean) {
        if (success) db?.setTransactionSuccessful()
        db?.endTransaction()
    }
}